### PR TITLE
ENH: Improve error mesage verbosity in string accessor

### LINKED ISSUE #59659: ENH: increase verbosity on error
### Feature Type

- [ ] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

On a specific type error the error causing type is not logged in the error

### Feature Description

By replacing:
`raise AttributeError("Can only use .str accessor with string values!")`
with
`raise AttributeError(f"Can only use .str accessor with string values! Inferred dType: {inferred_dtype}")`
the problem would be solved

Details can be found in [here](https://github.com/pandas-dev/pandas/pull/59649/files)

### Alternative Solutions

Use a debugger and set a break point

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by KevsterAmp**
- [x] closes #59659  (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Continued stale PR: #59649

### CODE REVIEW

**Comment by KevsterAmp** [comment]
@rhshadrach - could you take a look at this? thanks

File: pandas/core/strings/accessor.py [hunk 1]
```diff
@@ -255,7 +255,9 @@ def _validate(data):
         inferred_dtype = lib.infer_dtype(values, skipna=True)
 
         if inferred_dtype not in allowed_types:
-            raise AttributeError("Can only use .str accessor with string values!")
+            raise AttributeError(
+                f"Can only use .str accessor with string values!, not {inferred_dtype}"
```
Comment (rhshadrach) [inline pandas/core/strings/accessor.py:259]: Can you remove the `!`, I believe it is not grammatically correct.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looking good - if this isn't currently hit by tests, they should be added.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looks good! Can you update the test `test_str_attribute` in `tests/series/accessors/test_str_accessor.py`. Right now it is checking this case, but only checks part of the message. I think we can just update that to test the full message for this.

**Comment by rhshadrach** [comment]
I think this looks good, but the CI has many unrelated failures. I'd like to get those resolved before merging PRs.

**Comment by rhshadrach** [comment]
CI is fixed, can you merge main.

**Comment by KevsterAmp** [comment]
@rhshadrach - CI looks good. Thanks

**Review by rhshadrach — APPROVED** [review]
lgtm

**Pull request merged**